6.js

```diff
@@ -27,6 +32,71 @@ function addToCartButton(event) {
     }
 }
 
+function handleModalKeyDown(event) {
+    if (event.code === 'ArrowRight') {
+        $preview.innerHTML = "";
+        var imagePath = $thumbnails.lastChild.firstChild.dataset.original;
+        $img = document.createElement('img');
+        $img.src = imagePath;
+        $preview.appendChild($img);
+    }
+    if (event.code === 'ArrowLeft') {
+        $preview.innerHTML = "";
+        var imagePath = $thumbnails.firstChild.firstChild.dataset.original;
+        $img = document.createElement('img');
+        $img.src = imagePath;
+        $preview.appendChild($img);
+    }
+}
+function showBasketContent(event) {
+    $catalog.setAttribute('class', 'hidden');
+    event.target.setAttribute('class', 'hidden');
+    $delivery.previousElementSibling.classList.remove('hidden');
+    $basket.innerHTML = '';
+    var $list = document.createElement('ul');
+    $list.addEventListener('click', handleItemRemove);
+    $basket.removeEventListener('click', showBasketContent);
+    
+    for (var i = 0; i < basket.length; i++) {
+        var $li = document.createElement('li');
+        $li.textContent = basket[i].quantity + ' ' + basket[i].name + ' ' + basket[i].price * basket[i].quantity;
+        var $button = document.createElement('button');
+        $button.textContent = 'remove item';
+        $li.appendChild($button);
+        $list.appendChild($li);
+    }
+    $basket.appendChild($list);
+    $next1.addEventListener('click', showDeliveryOptions);
+}
+
+function showDeliveryOptions(event) {
+    $basket.setAttribute('class', 'hidden');
+    event.target.setAttribute('class', 'hidden');
+    $delivery.classList.remove('hidden');
+    $comment.previousElementSibling.classList.remove('hidden');
+    $next2.addEventListener('click', showComments);
+}
+
+function showComments(event) {
+    event.target.setAttribute('class', 'hidden');
+    $delivery.setAttribute('class', 'hidden');
+    $comment.classList.remove('hidden');
+}
+
+function handleItemRemove(event) {
+    if (event.target.tagName === 'BUTTON') {
+        $basketTotal.innerHTML = '';
+        for (var i = 0; i < basket.length; i++) {
+            if (event.target.parentNode.firstChild.textContent.split(' ')[1] === basket[i].name) {
+                basket[i].quantity > 0 ? (basket[i].quantity-- && (totalPrice -= basket[i].price) && totalItems--) : event.target.parentNode.parentNode.removeChild(event.target.parentNode);
+                event.target.parentNode.firstChild.textContent = basket[i].quantity + ' ' + basket[i].name + ' ' + basket[i].price * basket[i].quantity;  
+                (totalItems === 0) ? $basketTotal.textContent = 'Basket is empty' : $basketTotal.textContent = totalItems + '  items in basket for ' + totalPrice + '$';            
```

Comment:
Не совсем верно используете тернарник
```suggestion
                $basketTotal.textContent = (totalItems === 0) ? 'Basket is empty' : totalItems + '  items in basket for ' + totalPrice + '$';            
```